telegram/ext/builders.py

```diff
@@ -0,0 +1,1196 @@
+        self._request_kwargs: DVInput[Dict[str, Any]] = DefaultValue({})
+        self._request: ODVInput['Request'] = DEFAULT_NONE
+        self._private_key: ODVInput[bytes] = DEFAULT_NONE
+        self._private_key_password: ODVInput[bytes] = DEFAULT_NONE
+        self._defaults: ODVInput['Defaults'] = DEFAULT_NONE
+        self._arbitrary_callback_data: DVInput[Union[bool, int]] = DEFAULT_FALSE
+        self._bot: Bot = DEFAULT_NONE  # type: ignore[assignment]
+        self._update_queue: DVInput[Queue] = DefaultValue(Queue())
+        self._workers: DVInput[int] = DefaultValue(4)
+        self._exception_event: DVInput[Event] = DefaultValue(Event())
+        self._job_queue: ODVInput['JobQueue'] = DefaultValue(JobQueue())
+        self._persistence: ODVInput['BasePersistence'] = DEFAULT_NONE
+        self._context_types: DVInput[ContextTypes] = DefaultValue(ContextTypes())
+        self._dispatcher: ODVInput['Dispatcher'] = DEFAULT_NONE
+        self._user_signal_handler: Optional[Callable[[int, object], Any]] = None
+        self._dispatcher_class: DVInput[Type[Dispatcher]] = DefaultValue(Dispatcher)
+        self._dispatcher_kwargs: Dict[str, object] = {}
+        self._updater_class: Type[Updater] = Updater
+        self._updater_kwargs: Dict[str, object] = {}
+
+    def _get_connection_pool_size(self) -> int:
+        # For the standard use case (Updater + Dispatcher + Bot)
+        # we need a connection pool the size of:
+        # * for each of the workers
+        # * 1 for Dispatcher
+        # * 1 for Updater (even if webhook is used, we can spare a connection)
+        # * 1 for JobQueue
+        # * 1 for main thread
+        return DefaultValue.get_value(self._workers) + 4
+
+    def _build_ext_bot(self) -> ExtBot:
+        if isinstance(self._token, DefaultValue):
+            raise RuntimeError('No bot token was set.')
+        request_kwargs = DefaultValue.get_value(self._request_kwargs)
+        if (
+            self._request is DEFAULT_NONE
+            and 'con_pool_size' not in request_kwargs  # pylint: disable=E1135
+        ):
+            request_kwargs[  # pylint: disable=E1137
+                'con_pool_size'
+            ] = self._get_connection_pool_size()
+        return ExtBot(
+            token=self._token,
+            base_url=DefaultValue.get_value(self._base_url),
+            base_file_url=DefaultValue.get_value(self._base_file_url),
+            private_key=DefaultValue.get_value(self._private_key),
+            private_key_password=DefaultValue.get_value(self._private_key_password),
+            defaults=DefaultValue.get_value(self._defaults),
+            arbitrary_callback_data=DefaultValue.get_value(self._arbitrary_callback_data),
+            request=self._request  # type: ignore[arg-type]
+            if self._request is not DEFAULT_NONE
+            else Request(**request_kwargs),  # pylint: disable=E1134
```

Comment:
just a style comment: if you can't get the required value in one liner, better calculate it ahead. makes reading easier on the eye (especially that black has some weird preference not to indent the code here).  
Another reason would be because you only need to deduce what request_kwargs are iff you need to construct a request object.

i.e.
```python
request = self._request
if request is DEFAULT_NONE:
    request_kwargs = DefaultValue.get_value(self._request_kwargs)
    if "con_pool_size" not in request_kwargs:
        request_kwargs["con_pool_size"] = self._get_connection_pool_size()
    request = Request(**request_kwargs)

return ExtBot(
    .
    .
    request=request,
)